### ISSUE TITLE
warpx unit test with image check

### ISSUE THREAD

**Issue by nicolemarsaglia** [Alpine-DAV/ascent#1280]
We can render streamlines but the input dataset particles have too tiny a radius for the raytracer to pick up their presence. To get an image I manually scaled up the data. 

In order to do the necessary scaling from the unit test we should move it from tests/vtkh to tests/ascent to easily apply other pipelines. 

https://github.com/Alpine-DAV/ascent/blob/develop/src/tests/vtkh/t_vtk-h_warpx_streamline.cpp#L167

**Comment by nicolemarsaglia** [comment]
The "finally looking at this" updates/notes: I don't think we need to move it to tests/ascent. I don't see why we can't just use vtkh::scale or vtkh::transform's scale. Other issue is the serial test is complaining about no mpich, I don't think it's in the unit test, but maybe because vtkm is built with mpi it assumes it is parallel, tbd.

**Comment by nicolemarsaglia** [comment]
New development: using scale of vtkh::PointTransform is erasing all the fields except the original and new coords on the second domain. first domain looks good. This feels familiar.

**Comment by cyrush** [comment]
In the Ascent filter we have logic to pass all fields. Do we just need extra args in the multi domain vtk-h case?

**Comment by nicolemarsaglia** [comment]
Yeah, it's this function: https://github.com/Alpine-DAV/ascent/blob/bcd4536ed9e0061f847978dc564fc59e1c2b0006/src/libs/vtkh/filters/Filter.cpp#L109 

we could pass fields for each domain respectively rather than domain 0

**Comment by nicolemarsaglia** [comment]
what we want vs what I'm getting. This also feels familiar?  

![Image](https://github.com/user-attachments/assets/7ca5f796-a482-4a6d-8d3b-196dc4651d1d)
![Image](https://github.com/user-attachments/assets/b0a256d0-463f-4aed-829e-8fa8245abe04)
![Image](https://github.com/user-attachments/assets/4603dbf9-ab6a-47c0-8a3c-0239c187b76c)
![Image](https://github.com/user-attachments/assets/9f62bf11-6950-4eab-9510-8fcde3cb4d20)

**Comment by cyrush** [comment]
> Yeah, it's this function:
> 
> [ascent/src/libs/vtkh/filters/Filter.cpp](https://github.com/Alpine-DAV/ascent/blob/bcd4536ed9e0061f847978dc564fc59e1c2b0006/src/libs/vtkh/filters/Filter.cpp#L109)
> 
> Line 109 in [bcd4536](/Alpine-DAV/ascent/commit/bcd4536ed9e0061f847978dc564fc59e1c2b0006)
>  Filter::MapAllFields() 
> 
> we could pass fields for each domain respectively rather than domain 0

Yes -- this logic could be faulty if there are fields that exist on one domain, but not others.
What we really want is the unique set of field names across all domains.
We could loop over all domains and add the fields to a set (or only add to the vector if not already an entry)

**Comment by cyrush** [comment]
bottom image is what we expect?

Could it be the integral time is less in our unit test?

**Comment by nicolemarsaglia** [comment]
> bottom image is what we expect?
> 
> Could it be the integral time is less in our unit test?

Yeah, bottom is what we want. It is the output of the warpx filter. Top images are Ascent rendering that output with the vtkm tubes.

**Comment by nicolemarsaglia** [comment]
![Image](https://github.com/user-attachments/assets/e1d1c697-d9f3-450d-8e16-f560f993cc0e)

vtk file after applying the tubes. So Ascent isn't getting it wrong, it's the tubes!

**Comment by cyrush** [comment]
First one is close to correct? Hard to tell with different views and image resolutions.

**Comment by nicolemarsaglia** [comment]
No :( it's a low res version of the visit incorrect long-boi using tubes

**Comment by nicolemarsaglia** [comment]
Actually, I might be wrong. It might just be so low res I think it looks wrong. I've reduce how much I'm scaling the data and things are looking better. Scaling up is shortening my tubes so it has to do with the max length I'm passing into the warpx filter that is not standardized for my scaling! ahhhhhhhh literally figuring this out as I type. noice

![Image](https://github.com/user-attachments/assets/05b88e4c-1c0a-4294-90cb-209f61d74a6c)

**Comment by nicolemarsaglia** [comment]
![Image](https://github.com/user-attachments/assets/c0143d36-e3bd-463b-944f-05519dfac6c9)

turns out we dont' even need to use the scale transform. 

but of course now I have the idea to apply the scale AFTER the warpx filter to avoid the short tubes, but when I briefly tried it I got an error. will pick a little more and write issue at some point.

**Comment by nicolemarsaglia** [comment]
@cyrush  can you point me to where the cmake logic is for checking if vtkm mpi is on for the serial case?  

Would I do it in the tests? 
https://github.com/Alpine-DAV/ascent/blob/develop/src/tests/vtkh/CMakeLists.txt#L101

And if so, does it go under this block or do we need to have a if (VTKM_MPI_FOUND)?

**Comment by nicolemarsaglia** [comment]
> [@cyrush](https://github.com/cyrush) can you point me to where the cmake logic is for checking if vtkm mpi is on for the serial case?
> 
> Would I do it in the tests? https://github.com/Alpine-DAV/ascent/blob/develop/src/tests/vtkh/CMakeLists.txt#L101
> 
> And if so, does it go under this block or do we need to have a if (VTKM_MPI_FOUND)?

Wait, I guess something like this may be correct(?): https://github.com/Alpine-DAV/ascent/blob/develop/src/tests/vtkh/CMakeLists.txt#L70

**Comment by cyrush** [comment]
In https://github.com/Alpine-DAV/ascent/blob/develop/src/cmake/thirdparty/SetupVTKm.cmake

We want to check for `VTKm_ENABLE_MPI`  and make sure MPI is then used along side VTK-m even in serial.

Here is an example of how we support mfem:

https://github.com/Alpine-DAV/ascent/blob/29ca91b13b15328d8bd1272ea6054638072b3f90/src/cmake/thirdparty/SetupMFEM.cmake#L150

but this isn't quite correct, we may need something more like:

https://github.com/LLNL/conduit/blob/b4315b70b66a708b3f597dfc22063908f5ad26e3/src/libs/relay/CMakeLists.txt#L191

In the case that we use implicit MPI support (cray compilers)


We also want this cmake logic to translate into a new header def in:

https://github.com/Alpine-DAV/ascent/blob/develop/src/libs/ascent/ascent_config.h.in

`ASCENT_VTKM_MPI_ENABLED`

So if we need conditional compiles, we can check if it is defined.

**Comment by cyrush** [comment]
When compiled with MPI, I don't think the VTK-m path will work w/o MPI:

```
[ RUN      ] vtkh_serial_warpx_streamlines.vtkh_serial_warpx_streamlines
Running serial WarpX Charged Particle Advection
2025-05-14 11:46:57.851 (   0.004s) [        48F2EE00]VTKDataSetReaderBase.cx:304   WARN| Reader may not correctly read >v4.2 files. Reading version 5.1.

2025-05-14 11:46:57.852 (   0.004s) [        48F2EE00]VTKDataSetReaderBase.cx:498   INFO| Support for global field MeshName not implemented. Skipping.
2025-05-14 11:46:57.852 (   0.004s) [        48F2EE00]VTKDataSetReaderBase.cx:498   INFO| Support for global field CYCLE not implemented. Skipping.
2025-05-14 11:46:57.852 (   0.004s) [        48F2EE00]VTKDataSetReaderBase.cx:498   INFO| Support for global field TIME not implemented. Skipping.
2025-05-14 11:46:57.889 (   0.041s) [        48F2EE00]RuntimeDeviceInformatio:220   WARN| Initializing 'RuntimeDeviceConfigurations' with uninitialized configOptions. Did you call vtkm::cont::Initialize?
CFL length : 8.5258e-16
Attempting to use an MPI routine before initializing MPICH
```

We should disable this test when VTK-m has MPI support. 

To support this, we need 

https://github.com/Alpine-DAV/ascent/blob/develop/src/libs/ascent/ascent_config.h.in

ASCENT_VTKM_MPI_ENABLED

(or the VTKH analog since the test is in VTK-h)

**Comment by cyrush** [comment]
resolved in #1489
